pex/venv/pex.py

```diff
@@ -489,6 +489,12 @@ def sys_executable_paths():
                 os.execv(script_path, [script_path] + sys.argv[1:])
 
             pex_interpreter = pex_overrides.get("PEX_INTERPRETER", "").lower() in ("1", "true")
+
+            if pex_interpreter:
```

Comment:
I think this needs to actually test `entry_point == PEX_INTERPRETER_ENTRYPOINT` down further. If the venv pex was built with no entrypoint, then `{entry_point!r}` will evaluate to the `None` literal and the `entry_point` will be assigned `code:interact` which you'll have missed as things stand currently.

Reply:
Ah right, I forgot about the non-env-var case.